packages/file_selector/file_selector_web/lib/src/utils.dart

```diff
@@ -0,0 +1,43 @@
+import 'dart:html';
+import 'package:file_selector_platform_interface/file_selector_platform_interface.dart';
+
+/// Convert list of XTypeGroups to a comma-separated string
+String acceptedTypesToString(List<XTypeGroup> acceptedTypes) {
+  if (acceptedTypes == null) return '';
+  final List<String> allTypes = [];
+  for (final group in acceptedTypes) {
+    _assertTypeGroupIsValid(group);
+    if (group.extensions != null) {
+      allTypes.addAll(group.extensions.map(_normalizeExtension));
+    }
+    if (group.mimeTypes != null) {
+      allTypes.addAll(group.mimeTypes);
+    }
+    if (group.webWildCards != null) {
+      allTypes.addAll(group.webWildCards);
+    }
+  }
+  return allTypes.join(',');
+}
+
+/// Helper to convert an html.File to an XFile
+XFile convertFileToXFile(File file) => XFile(
+      Url.createObjectUrl(file),
+      name: file.name,
+      length: file.size,
+      lastModified: DateTime.fromMillisecondsSinceEpoch(file.lastModified),
+    );
+
```

Comment:
(If you extract this `convertFileToXFile` to an html_utils.dart file, you can test the rest of utils.dart with a normal unit test, it seems?)